lib.py

```diff
@@ -303,6 +313,36 @@ def process_run_result(
     return result
 
 
+async def try_answer(
+    session: http.ClientSession,
+    day: int,
+    part: int,
+    answer_key: str,
+    answer: int | str,
+    answers_map: dict,
+):
+    """Try an answer with the AOC website."""
+    current_year = year()
+    answer_url = f"https://adventofcode.com/{current_year}/day/{day}/answer"
+    payload = {
+        "level": part,
+        "answer": answer,
+    }
+
+    token = settings.aoc_auth.tokens[answer_key]
+
+    async with session.post(answer_url, data=payload, cookies={"session": token}) as response:
+        response_text = await response.text()
+        if "That's not the right answer" in response_text:
+            return False
+        elif "That's the right answer" in response_text:
+            # We have a correct answer
+            answers_map[answer_key] = answer
+            return True
+        else:
+            logger.error("Unrecognized response from AOC website:\n%s", response_text)
```

Comment:
There's a pretty huge gap here: If an answer is queried, but doesn't make it to the DB, the website will no longer give us that answer via the URL we're using. Instead, we have to parse the raw problem page (eg, `https://adventofcode.com/2023/day/12`) in order to find the answers we've already unlocked. 

The work-around is easy enough, just manually load the appropriate answer into the DB.

We will almost certainly bump into this limitation eventually. The question is _how often_ we'll hit the issue.